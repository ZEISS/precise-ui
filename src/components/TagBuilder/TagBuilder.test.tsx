import * as React from 'react';
import * as enzyme from 'enzyme';
import { TagBuilderInt } from './TagBuilderInt.part';

const tags = ['tags', 'list', 'goes', 'here'];

describe('<TagBuilderInt />', () => {
  it('should render the component', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not be in controlled mode by default', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt />);
    expect(wrapper.state('controlled')).toBe(false);
  });

  it('should not be controlled with defaultValue set', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt defaultValue={tags} />);
    expect(wrapper.state('controlled')).toBe(false);
  });

  it('should be in controlled mode when value prop set', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt value={tags} />);
    expect(wrapper.state('controlled')).toBe(true);
  });

  it('should render component with empty value', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt info="info block" value={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with info block and value', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt info="info block" value={tags} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with error block and default value', () => {
    const wrapper = enzyme.shallow(<TagBuilderInt error="error block" defaultValue={tags} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('default input position should be undefined if just value set', () => {
    const wrapper = enzyme.mount(<TagBuilderInt value={tags} />);
    expect(wrapper.state('inputPosition')).toBe(undefined);
  });

  it('default input position should be undefined if just defaultValue string', () => {
    const wrapper = enzyme.mount(<TagBuilderInt defaultValue={tags} />);
    expect(wrapper.state('inputPosition')).toBe(undefined);
  });

  it('default input position should be undefined when no data is set', () => {
    const wrapper = enzyme.mount(<TagBuilderInt />);
    expect(wrapper.state('inputPosition')).toBe(undefined);
  });
});
