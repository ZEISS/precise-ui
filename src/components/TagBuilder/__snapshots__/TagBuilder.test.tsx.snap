// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagBuilderInt /> should render component with empty value 1`] = `
<styled.div>
  <styled.div
    border={1}
    focused={false}
    hasValue={false}
  >
    <Component
      error={false}
      focused={false}
      hasValue={false}
      onClick={[Function]}
    >
      <styled.div
        labelShown={false}
        tagRenderer={false}
      >
        <styled.div
          key="input"
          onKeyDown={[Function]}
          tagRenderer={false}
        >
          <styled.input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            valid={true}
            value=""
          />
        </styled.div>
      </styled.div>
    </Component>
    <Component
      hasValue={false}
    />
  </styled.div>
  <Component
    bottom="xsmall"
    left="medium"
    top="xsmall"
  >
    <Component
      info="info block"
    />
  </Component>
</styled.div>
`;

exports[`<TagBuilderInt /> should render component with error block and default value 1`] = `
<styled.div>
  <styled.div
    border={3}
    focused={false}
    hasValue={true}
  >
    <Component
      error={true}
      focused={false}
      hasValue={true}
      onClick={[Function]}
    >
      <styled.div
        labelShown={false}
        tagRenderer={false}
      >
        <Styled(styled.span)
          key="tags0"
        >
          <styled.span>
            tags
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="list1"
        >
          <styled.span>
            list
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="goes2"
        >
          <styled.span>
            goes
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="here3"
        >
          <styled.span>
            here
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <styled.div
          key="input"
          onKeyDown={[Function]}
          tagRenderer={false}
        >
          <styled.input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            valid={true}
            value=""
          />
        </styled.div>
      </styled.div>
    </Component>
    <Component
      error="error block"
      hasValue={true}
    />
  </styled.div>
  <Component
    bottom="xsmall"
    left="medium"
    top="xsmall"
  >
    <Component
      error="error block"
    />
  </Component>
</styled.div>
`;

exports[`<TagBuilderInt /> should render component with info block and value 1`] = `
<styled.div>
  <styled.div
    border={1}
    focused={false}
    hasValue={true}
  >
    <Component
      error={false}
      focused={false}
      hasValue={true}
      onClick={[Function]}
    >
      <styled.div
        labelShown={false}
        tagRenderer={false}
      >
        <Styled(styled.span)
          key="tags0"
        >
          <styled.span>
            tags
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="list1"
        >
          <styled.span>
            list
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="goes2"
        >
          <styled.span>
            goes
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <Styled(styled.span)
          key="here3"
        >
          <styled.span>
            here
          </styled.span>
          <CloseIcon
            onClick={[Function]}
            onMouseDown={[Function]}
          />
        </Styled(styled.span)>
        <styled.div
          key="input"
          onKeyDown={[Function]}
          tagRenderer={false}
        >
          <styled.input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            valid={true}
            value=""
          />
        </styled.div>
      </styled.div>
    </Component>
    <Component
      hasValue={true}
    />
  </styled.div>
  <Component
    bottom="xsmall"
    left="medium"
    top="xsmall"
  >
    <Component
      info="info block"
    />
  </Component>
</styled.div>
`;

exports[`<TagBuilderInt /> should render the component 1`] = `
<styled.div>
  <styled.div
    border={1}
    focused={false}
    hasValue={false}
  >
    <Component
      error={false}
      focused={false}
      hasValue={false}
      onClick={[Function]}
    >
      <styled.div
        labelShown={false}
        tagRenderer={false}
      >
        <styled.div
          key="input"
          onKeyDown={[Function]}
          tagRenderer={false}
        >
          <styled.input
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            valid={true}
            value=""
          />
        </styled.div>
      </styled.div>
    </Component>
    <Component
      hasValue={false}
    />
  </styled.div>
</styled.div>
`;
